compress/compress_private.h

```diff
@@ -24,12 +25,25 @@
 #define MUTT_COMPRESS_COMPRESS_PRIVATE_H
 
 #define COMPRESS_OPS(_name)                  \
-  const struct ComprOps compr_##_name##_ops = {     \
-    .name       = #_name,                           \
-    .open       = compr_##_name##_open,             \
-    .compress   = compr_##_name##_compress,         \
-    .decompress = compr_##_name##_decompress,       \
-    .close      = compr_##_name##_close,            \
+  const struct ComprOps compr_##_name##_ops = {        \
+    .name            = #_name,                         \
+    .open            = compr_##_name##_open,           \
+    .set_level       = compr_##_name##_set_level,      \
+    .set_dictionary  = NULL,                           \
+    .compress        = compr_##_name##_compress,       \
+    .decompress      = compr_##_name##_decompress,     \
+    .close           = compr_##_name##_close,          \
+  };
+
+#define COMPRESS_DICT(_name)                  \
```

Comment:
I'd prefer to have only one macro, and explictely define the `_set_dictionary()` function as a no-op for backends that do not support it.

Reply:
How does this work? Could you show me your suggestion ?

Reply:
I thing I got it, see 6744fc74eb439daf1dd9d2e3e361204f3ba8ca41 ;)